src/main/java/org/opensearch/knn/index/IndexUtil.java

```diff
@@ -51,6 +51,8 @@ public class IndexUtil {
             put(MODEL_NODE_ASSIGNMENT_KEY, MINIMAL_SUPPORTED_VERSION_FOR_MODEL_NODE_ASSIGNMENT);
             put(MODEL_METHOD_COMPONENT_CONTEXT_KEY, MINIMAL_SUPPORTED_VERSION_FOR_MODEL_METHOD_COMPONENT_CONTEXT);
             put(KNNConstants.RADIAL_SEARCH_KEY, MINIMAL_SUPPORTED_VERSION_FOR_RADIAL_SEARCH);
+            // TODO: Make 2.15 when we merge into 2.x. Making it 3.0.0 to make sure BWC passes
+            put(KNNConstants.EF_SEARCH, Version.V_3_0_0);
```

Comment:
nit: MINIMAL_SUPPORTED_VERSION_FOR_EF_SEARCH_AS_QUERY_PARAM

Reply:
I saw that and it felt redundant considering we already have one constant and there is no reuse anticipated. I can change if you feel strongly about it

Reply:
In this way you can only flip that value in 2.x and main at constant value instead of touching this method. what do you think?

Reply:
The map itself is the constant, doesn't seem to be a big deal here. either way works for me